src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3205,3 +3218,234 @@ def _aks_approuting_update(
         return None
 
     return aks_update_decorator.update_mc(mc)
+
+
+def _aks_run_command(
+        cmd,
+        vm_set_type,
+        managed_resource_group,
+        vmss_name=None,
+        instance_id=None,
+        vm_name=None,
+        fqdn=None,
+        custom_endpoints=None):
+    try:
+        command = f"bash /opt/azure/containers/aks-check-network.sh {fqdn}"
+        if custom_endpoints:
+            endpoint_list = [endpoint.strip() for endpoint in custom_endpoints.split(",")]
+            all_endpoints = ",".join(endpoint_list)
+            command += f" {all_endpoints}"
+            logger.debug("Full command: %s", command)
+
+        compute_client = get_compute_client(cmd.cli_ctx)
+
+        if vm_set_type == "VirtualMachineScaleSets":
+            RunCommandInput = cmd.get_models('RunCommandInput',
+                                             resource_type=ResourceType.MGMT_COMPUTE,
+                                             operation_group="virtual_machine_scale_sets")
+            command_result = LongRunningOperation(cmd.cli_ctx)(
+                compute_client.virtual_machine_scale_set_vms.begin_run_command(
+                    managed_resource_group, vmss_name, instance_id,
+                    RunCommandInput(command_id="RunShellScript", script=[command])))
+        elif vm_set_type == "AvailabilitySet":
```

Comment:
what if the vm type is not one of vmss or vmas?

Reply:
You're right. I just added the last `else`